pkg/cloud/azure/services/virtualmachines/virtualmachines.go

```diff
@@ -304,6 +304,13 @@ func (s *Service) deriveVirtualMachineParameters(vmSpec *Spec, nic network.Inter
 		}
 	}
 
+	if virtualMachine.StorageProfile.ImageReference.Sku != nil {
```

Comment:
Is there a difference between a `nil` plan and a non-nil plan with `nil` entries?

Could we make sure to add an appropriate comment as to why this is required

Reply:
> Is there a difference between a `nil` plan and a non-nil plan with `nil` entries?

I think your question is really referring to the `ImageReference` not the `Plan`. If you were really asking about the `Plan` that we are creating below, I'm not sure exactly how the `Plan` is used further upstream. The only [further usage of `Plan`](https://github.com/Azure/azure-sdk-for-go/blob/main/services/compute/mgmt/2021-03-01/compute/models.go#L14544-L14546) I see just uses whether `Plan` is nil to determine whether it is serialized.

But I think your question is about `ImageReference` being `nil`. In OpenShift `imageReference` will always be populated; the field values should be mutually exclusive and either only an `ID` or `Publisher`, `Offer`, `SKU` & `Version`. 

I'm not familiar to what extent the CAPZ does validation, so do we want to build in more checking? Something to the effect of:
```go
ir := virtualMachine.StorageProfile.ImageReference
usePlan := ir.Sku != nil && ir.Publisher != nil && ir.Offer != nil && ir.Version != nil && ir.ID == nil
if usePlan {
```

Reply:
I was just thinking if we could get rid of this if statement, it's a bit of a nit so no worries